### PR TITLE
Chore/database functions v2

### PR DESCRIPTION
**Pull request by joshenlim**
functions-assistant

### CODE REVIEW

File: apps/studio/components/ui/AIAssistantPanel/AIAssistant.tsx [hunk 1]
```diff
@@ -0,0 +1,642 @@
...
+                      <AlertDescription_Shadcn_ className="flex flex-col gap-4">
+                        <span>
+                          Quick reminder that you're not sending project metadata with your queries.
+                          By opting into sending anonymous data, Supabase AI can improve the answers
+                          it shows you.
+                        </span>
+                        <Button
+                          type="default"
+                          className="w-fit"
+                          // onClick={() => setIsConfirmOptInModalOpen(true)}
```
Comment (alaister) [inline apps/studio/components/ui/AIAssistantPanel/AIAssistant.tsx:308]: Still needed?
```suggestion
```

File: apps/studio/components/ui/AIAssistantPanel/AIAssistantPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+
+  const { open, editor, content } = aiAssistantPanel
+  const previousEditor = usePrevious(editor)
+
+  const [isAcknowledged, setIsAcknowledged] = useLocalStorage(
+    LOCAL_STORAGE_KEYS.SQL_SCRATCH_PAD_BANNER_ACKNOWLEDGED,
+    false
+  )
+
+  const [chatId, setChatId] = useState(uuidv4())
```
Comment (alaister) [inline apps/studio/components/ui/AIAssistantPanel/AIAssistantPanel.tsx:60]: Here to micro-optimise all the things 😅
```suggestion
  const [chatId, setChatId] = useState(() => uuidv4())
```

File: apps/studio/components/ui/AIAssistantPanel/EntitiesDropdownMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+}: EntitiesDropdownMenuProps) => {
+  const project = useSelectedProject()
+  const [search, setSearch] = useState('')
+
+  // [Joshen] Not gonna do infinite loading here, but rather just try to load sufficient tables
+  // I reckon users will opt to just search if there's many tables here given the small real estate
+  const {
+    data: entitiesData,
+    isLoading,
+    isFetching,
```
Comment (alaister) [inline apps/studio/components/ui/AIAssistantPanel/EntitiesDropdownMenu.tsx:36]: Add a cheeky error state?

File: apps/studio/components/ui/AIAssistantPanel/Message.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+export const Message = memo(function Message({
+  name,
+  role,
+  content,
+  createdAt,
+  isDebug,
+  isSelected = false,
+  context,
+  children,
+  action = <></>,
```
Comment (alaister) [inline apps/studio/components/ui/AIAssistantPanel/Message.tsx:35]: nit
```suggestion
  action = null,
```

**Review by alaister — APPROVED** [review]
LGTM!

- Tested on preview
  - Tested old editor to ensure functionality
  - Tested new assistant

Most of the comments are nits:

File: apps/studio/components/ui/AIAssistantPanel/AIAssistant.tsx (hunk 1 above)
Comment (joshenlim) [reply apps/studio/components/ui/AIAssistantPanel/AIAssistant.tsx:308]: oh good catch this! i need to set up if the user didnt opt into sending metadata

**Pull request merged**